### PR TITLE
modernize and cleanup theme

### PR DESCRIPTION
**Pull request by martymcguire**
@dshanske's updates to bring the microformats.org theme up from WP 2.5

some WordPress APIs have changed or been deprecated and this will require some migration steps.

Let's update this PR with those as we find them!

* google analytics account info has been moved to a wp_config.php variable. be sure to set `MICROFORMATS_GOOGLE_ANALYTICS`.
* deprecated link system was being used for a menu. updating to the current WP menu system requires configuring a menu that has the right menu options in the backend.
  * from here: "Microformats.org Main Menu" link category /wp-admin/edit-tags.php?taxonomy=link_category
  * links:
    * Blog /blog
    * Wiki /wiki
    * Discuss /wiki/irc
    * About /wiki/about
    * Code & Tools /wiki/code-tools
    * Get Started /wiki/get-started
* replace with a new menu: /wp-admin/nav-menus.php
  * remember to make it primary: http://microformats.vm/wp-admin/nav-menus.php?action=locations
  * ~~fixme: styles don't quite match~~ fix'd
* ~~sidebar needs to be customized to match prod:~~ (no longer req'd)
    * image
* ... more?

### CODE REVIEW

**Comment by dshanske** [comment]
Suggest overall rule of thumb is to replace custom coded sections with WordPress functions when such functions were introduced post-2.0

* Uses deprecated functions(many deprecated prior to 2.8.0) - To be repelaced with current equivalents
* Hardcodes Google Analytics and Technorati links (To Be Removed)
* Generates menu using the links/bookmarks system, now deprecated and also hardcoded to a specific category ID. Replace with the WordPress menu system...requires a menu with the current links be generated, but removes a lot of customization.
* Archive title generation done manually, replaced with title generating function introduced subsequently
* Direct includes of template files no longer permitted - replaced with call to get_search_form or get_template_part
* Split reusable pieces into their own files, called by get_template_part
* Remove custom coded comment template and replace with wp_list_comment built in function introduced in 2.7 with some microformats2 customizations
* Replace navigation elements with WordPress functions to generate equivalent navigation elements.
* Remove hard coded RSS link as this is now dynamically added to head. (Add wp_head to allow this...also allows WordPress plugins to function).
* Add WordPress Privacy functionality support adding during the period of GDPR interest.
* Update to HTML5 as appropriate...theme predates this.
* Internationalization for strings...WordPress allows for translations to be submitted as needd through their system
* Add proper mf2 markup. Decide if classic mf should be preserved or replaced.

**Comment by martymcguire** [comment]
@dshanske seeing some markup differences with doing the header menu as a menu rather than link category.

original markup:

```
<ul id="nav">
<li class="active"><a href="/blog" title="Microformats.org Home">Blog</a></li>
<li><a href="/wiki/" title="The microformats.org documentation and specifications">Wiki</a></li>
<li><a href="/wiki/irc" title="Join and read discussion forum to discuss and develop microformats">Discuss</a></li>
<li><a href="/wiki/about" title="What are microformats?">About</a></li>
<li><a href="/wiki/code-tools" title="Snippets and tools to help you publish microformats">Code &amp; Tools</a></li>
<li><a href="/wiki/get-started" title="A little guide on how to publish your first hCard">Get Started</a></li>
</ul>
```

new:

```
<nav id="nav">
<div class="menu-primary-container"><ul id="menu-primary" class="primary-menu"><li id="menu-item-504" class="menu-item menu-item-type-post_type menu-item-object-page menu-item-504"><a href="http://microformats.vm/blog">Blog</a></li>
<li id="menu-item-509" class="menu-item menu-item-type-custom menu-item-object-custom menu-item-509"><a href="/wiki">Wiki</a></li>
<li id="menu-item-510" class="menu-item menu-item-type-custom menu-item-object-custom menu-item-510"><a href="/wiki/irc">Discuss</a></li>
<li id="menu-item-511" class="menu-item menu-item-type-custom menu-item-object-custom menu-item-511"><a href="/wiki/about">About</a></li>
<li id="menu-item-512" class="menu-item menu-item-type-custom menu-item-object-custom menu-item-512"><a href="/wiki/code-tools">Code &amp; Tools</a></li>
<li id="menu-item-513" class="menu-item menu-item-type-custom menu-item-object-custom menu-item-513"><a href="/wiki/get-started">Get Started</a></li>
</ul></div>
</nav>
```

additionally, the blog link doesn't get marked as "active" in the new markup. :thinking:

**Comment by dshanske** [comment]
@martymcguire Fixed the active link issue and removed the div.

**Comment by dshanske** [comment]
The analytics code is restored, but the actual site specific code will be moved into wp-config so that the repo doesn't have it.

File: style.css [hunk 1]
```diff
@@ -183,19 +183,23 @@ address {
...
+#nav li.current-menu-item {
 	background: url(img/nav-top.gif) no-repeat top left;
 	}
-#nav li.active a {
+#nav li.current-menu-item a {
 	color: #333;
 	background: url(img/nav-bottom.gif) no-repeat bottom right !important;
 	}
 
+.menu-primary {
```
Comment (martymcguire) [inline style.css:197]: looks like this should be `.primary-menu` instead!

Comment (martymcguire) [reply style.css:197]: i'll update after merging!

**Pull request merged**